saos-search/src/main/java/pl/edu/icm/saos/search/analysis/solr/XSettingsFacetQueryApplier.java

```diff
@@ -0,0 +1,99 @@
+package pl.edu.icm.saos.search.analysis.solr;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Resource;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.common.params.FacetParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import pl.edu.icm.saos.search.analysis.request.XField;
+import pl.edu.icm.saos.search.analysis.request.XRange;
+import pl.edu.icm.saos.search.analysis.request.XSettings;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+/**
+ * @author madryk
+ */
+@Service
+public class XSettingsFacetQueryApplier {
+
+    private Map<XField, String> fieldNamesMappings = Maps.newHashMap();
+    
+    private List<XRangeConverter> xRangeConverters = Lists.newArrayList();
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    public void applyXSettingsToQuery(SolrQuery query, XSettings xSettings) {
+        
+        XField xField = xSettings.getField();
+        String fieldName = fieldNamesMappings.get(xField);
+        
+        if (fieldName == null) {
+            throw new IllegalArgumentException("No field mapping found for " + xField.name());
+        }
+        
+        XRange xRange = xSettings.getRange();
+        XRangeConverter xRangeConverter = pickApplicableRangeConverter(xRange);
```

Comment:
It's better to create a separate manager service because you have a separation of concerns and more concise testing classes. So here I would go with: XRangeConverterManager.getConverter(xRange)